runtime/service/src/main/java/org/apache/polaris/service/auth/JWTBroker.java

```diff
@@ -157,16 +143,18 @@ public TokenResponse generateFromClientSecrets(
     if (principal.isEmpty()) {
       return new TokenResponse(OAuthTokenErrorResponse.Error.unauthorized_client);
     }
-    String tokenString = generateTokenString(clientId, scope, principal.get().getId());
+    String tokenString =
+        generateTokenString(principal.get().getName(), principal.get().getId(), clientId, scope);
     return new TokenResponse(
         tokenString, TokenType.ACCESS_TOKEN.getValue(), maxTokenGenerationInSeconds);
   }
 
-  private String generateTokenString(String clientId, String scope, Long principalId) {
+  private String generateTokenString(
+      String principalName, long principalId, String clientId, String scope) {
     Instant now = Instant.now();
     return JWT.create()
         .withIssuer(ISSUER_KEY)
-        .withSubject(String.valueOf(principalId))
+        .withSubject(principalName)
```

Comment:
[for my understanding] we don't use principalId for lookups from metastore from sub and always use the prinicipal id from the claim ? if yes how do we plan to leverage principalName from the subject here ?

Reply:
Yes, exactly. Some `Authenticator`s could enforce by-name lookup since principal names must be unique.